model/project_ref.go

```diff
@@ -3299,7 +3299,7 @@ func ValidateTriggerDefinition(definition patch.PatchTriggerDefinition, parentPr
 		return definition, errors.Wrapf(err, "finding child project '%s'", definition.ChildProject)
 	}
 
-	if !utility.StringSliceContains([]string{"", AllStatuses, evergreen.LegacyPatchSucceeded, evergreen.VersionSucceeded, evergreen.VersionFailed}, definition.Status) {
+	if !utility.StringSliceContains([]string{"", AllStatuses, evergreen.VersionSucceeded, evergreen.VersionFailed}, definition.Status) {
```

Comment:
Not an action but would it be worth having some slice that encapsulates this? We don't add statuses (I think) so I think it's fine if this is the sole check but an idea

Reply:
yeah this could be a function constant that helps us know what we're checking anyway, like `validTriggerStatuses`